diff --git a/config.def.h b/config.def.h
index 1c0b587..38c3d46 100644
--- a/config.def.h
+++ b/config.def.h
@@ -58,6 +58,9 @@ static const Layout layouts[] = {
 static char dmenumon[2] = "0"; /* component of dmenucmd, manipulated in spawn() */
 static const char *dmenucmd[] = { "dmenu_run", "-m", dmenumon, "-fn", dmenufont, "-nb", col_gray1, "-nf", col_gray3, "-sb", col_cyan, "-sf", col_gray4, NULL };
 static const char *termcmd[]  = { "st", NULL };
+static const char *volume_up[] = { "amixer", "set", "Master", "5%+", NULL };
+static const char *volume_down[] = { "amixer", "set", "Master", "5%-", NULL };
+static const char *volume_mute[] = { "amixer", "set", "Master", "toggle", NULL };
 
 static Key keys[] = {
 	/* modifier                     key        function        argument */
@@ -84,6 +87,14 @@ static Key keys[] = {
 	{ MODKEY,                       XK_period, focusmon,       {.i = +1 } },
 	{ MODKEY|ShiftMask,             XK_comma,  tagmon,         {.i = -1 } },
 	{ MODKEY|ShiftMask,             XK_period, tagmon,         {.i = +1 } },
+	{ 0,                            XF86XK_MonBrightnessUp,    setbrightness, {.i = +1 } },
+	{ 0,                            XF86XK_MonBrightnessDown,  setbrightness, {.i = -1 } },
+	{ 0,                            XF86XK_AudioMute,          spawn, {.v = volume_mute } },
+	{ 0,                            XF86XK_AudioLowerVolume,   spawn, {.v = volume_down } },
+	{ 0,                            XF86XK_AudioRaiseVolume,   spawn, {.v = volume_up } },
 	TAGKEYS(                        XK_1,                      0)
 	TAGKEYS(                        XK_2,                      1)
 	TAGKEYS(                        XK_3,                      2)
diff --git a/dwm.c b/dwm.c
index b0b3466..da95d27 100644
--- a/dwm.c
+++ b/dwm.c
@@ -36,6 +36,7 @@
 #include <X11/Xlib.h>
 #include <X11/Xproto.h>
 #include <X11/Xutil.h>
+#include <X11/XF86keysym.h>
 #ifdef XINERAMA
 #include <X11/extensions/Xinerama.h>
 #endif /* XINERAMA */
@@ -200,6 +201,8 @@ static void sendmon(Client *c, Monitor *m);
 static void setclientstate(Client *c, long state);
 static void setfocus(Client *c);
 static void setfullscreen(Client *c, int fullscreen);
+static void setbrightness(const Arg *arg);
 static void setlayout(const Arg *arg);
 static void setmfact(const Arg *arg);
 static void setup(void);
@@ -1498,6 +1501,28 @@ setfullscreen(Client *c, int fullscreen)
 	}
 }
 
+void
+setbrightness(const Arg *arg)
+{
+	int brightness;
+	FILE *fd = fopen("/sys/class/backlight/intel_backlight/brightness", "w+");
+
+	fscanf(fd, "%d", &brightness);
+	brightness += (arg->i == 1) ? +500 : -500;
+	fprintf(fd, "%d", brightness);
+	fclose(fd);
+}
+
 void
 setlayout(const Arg *arg)
 {
